src/pickledb.rs

```diff
@@ -71,6 +72,34 @@ impl PickleDb {
         }
     }
 
+    /// Attempts to load a DB from a file, and if it fails to load, constructs a new `PickleDb` instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `db_path` - a path where the DB will be stored
+    /// * `dump_policy` - an enum value that determines the policy of dumping DB changes into the file. Please see
+    ///    [PickleDb::load()](#method.load) to understand the different policy options
+    /// * `serialization_method` - the serialization method to use for storing the data to memory and file
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use pickledb::{PickleDb, PickleDbDumpPolicy, SerializationMethod};
+    ///
+    /// let mut db = PickleDb::load_or_new("example.db", PickleDbDumpPolicy::AutoDump, SerializationMethod::Json);
+    /// ```
+    ///
+    pub fn load_or_new<P: AsRef<Path> + Clone>(
+        db_path: P,
+        dump_policy: PickleDbDumpPolicy,
+        serialization_method: SerializationMethod,
+    ) -> PickleDb {
+        match PickleDb::load(db_path.clone(), dump_policy, serialization_method) {
```

Comment:
I think the case where the path exists and fails to load should result in a load error. We can also avoid calling `load` when the path doesn't exist.

```rust
pub fn load_or_new<P: AsRef<Path> + Clone>(
    db_path: P,
    dump_policy: PickleDbDumpPolicy,
    serialization_method: SerializationMethod,
) -> Result<PickleDb> {
    match db_path.as_ref().exists() {
        true => PickleDb::load(db_path, dump_policy, serialization_method),
        false => Ok(PickleDb::new(db_path, dump_policy, serialization_method)),
    }
}
```

Reply:
This also removes the `Clone` trait bound, and will only require `P: AsRef<Path>`, the same as `load()` and `new()`